### PR TITLE
ARTEMIS-2285 Sorting by user column in connections view not working

### PR DESCRIPTION
**Pull request by sebthom**
The value returned by ConnectionsView.getField(con, "users") currently returns a Set which does not implement the Comparable interface thus ActiveMQAbstractView#getComparator() always returns a comparator which evaluates to 0.

### CODE REVIEW

**Comment by gaohoward** [comment]
looking good. Could you add a unit test?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java [hunk 1]
```diff
@@ -83,12 +83,12 @@ public Object getField(RemotingConnection connection, String fieldName) {
...
             return connection.getRemoteAddress();
          case "users":
-            Set<String> users = new HashSet<>();
+            Set<String> users = new TreeSet<>();
             for (ServerSession session : sessions) {
                String username = session.getUsername() == null ? "" : session.getUsername();
                users.add(username);
             }
-            return users;
+            return StringUtil.joinStringList(users, ",");
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java]: Whilst i understand treeset to give ordering by insert, the return shouldn't change the type, e.g just return users still. Or im missing something.... i guess a test may help to describe/show why its needed.

**Comment by sebthom** [comment]
I will try to provide a test case. stay tuned :)

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java (hunk 1 above)
Comment (sebthom) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java]: The returned value is a set of users of a single connection. to sort the connections by their user(s), these lists need to be comparable with each other. however treeset does not implement the comparable interface, so you cannot sort multiple treesets by there values. thus the PR now returns a comparable/sortable string representation of the set. does that make sense?

Comment (sebthom) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java]: The returned value is a set of users of a single connection. to sort the connections by their user(s), these lists need to be comparable with each other. however treeset does not implement the comparable interface, so you cannot sort multiple treesets by there values. thus the PR now returns a comparable/sortable string representation of the set. does that make sense?

**Comment by sebthom** [comment]
@gaohoward @michaelandrepearce  I added a test as requested.

**Comment by michaelandrepearce** [comment]
@sebthom thanks for this great contributions. merged.

**Pull request closed without merging**